### PR TITLE
feat: google analytics with advanced consent

### PR DESCRIPTION
**Pull request by MicLeey**
<details>
  <summary>Desktop:</summary>
  
  
  
</details>

<details>
  <summary>Tablet:</summary>
  
  

  
</details>

<details>
  <summary>Mobile:</summary>
  
  

</details>

Blockers:
- Let me know if copy should be adjusted
- I don't know if we need to let users granuarly opt out of GA metrics
- We need a cookie policy for GDPR and PECR
- A privacy policy is required for CCPA

### CODE REVIEW

**Comment by BlueAcidFrog** [comment]
FYI, I got an error message (google search console) describing a coverage problem: 

Coverage problems detected on community.ory.sh & ory.sh

To the owner of community.ory.sh:
There are 1 coverage type issues detected in Search Console for your website:
Main Error

Errors can prevent your page or feature from appearing in search results. The following errors were found on your website:
Sent URL is a soft 404 error.

We recommend that you fix these issues, if possible, so that your website provides the best possible experience and coverage in Google search.

**Comment by MicLeey** [comment]
It is locked behind an `envIsProduction`. Do we have a variable for staging/netlify? I could add that.

Plausible and Matomo are being loaded as per usual as they are using cookieless mode. Nothing changed for those

**Comment by aeneasr** [comment]
~I'm not sure, but I assume previes in netlify should set NODE_ENV to prod?~

> By default, Netlify’s buildbot sets NODE_ENV to development. You can change this value by setting a NODE_ENV environment variable.

https://docs.netlify.com/configure-builds/manage-dependencies/#node-js-environment

Will this build with NODE_ENV=production? Or do we have some dev dependencies that are needed to build?

**Comment by MicLeey** [comment]
It will - the same check is currently used to activate plausible in prod. Of course that means the actual GA tracking is untested. The banner works - screenshots in linked issue.

This is still blocked regardless, by a privacy/cookie policy for CCPA/GDPR respectively

**Comment by aeneasr** [comment]
Got it, thanks! We should probably update:

https://github.com/ory/web/blob/44f7a9cf74be255701ae1396bb2bd0726f18878b/.github/workflows/publish.yml#L29

https://github.com/ory/web/blob/44f7a9cf74be255701ae1396bb2bd0726f18878b/.github/workflows/publish.yml#L84

and add `NODE_ENV=production` to both!

File: static/_headers [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
   X-Frame-Options: SAMEORIGIN
   X-XSS-Protection: 1
-  Content-Security-Policy: default-src 'self' blob: 'unsafe-inline' 'unsafe-eval' unpkg.com *.algolianet.com *.algolianet.net storage.googleapis.com github.com *.githubusercontent.com gh-card.dev *.ory.sh *.youtube.com *.youtube-nocookie.com fonts.googleapis.com fonts.gstatic.com data: s.ytimg.com *.cloudfront.net; img-src * data:
+  Content-Security-Policy: default-src 'self' blob: 'unsafe-inline' 'unsafe-eval' unpkg.com *.algolianet.com *.algolianet.net storage.googleapis.com github.com *.githubusercontent.com gh-card.dev *.ory.sh *.youtube.com *.youtube-nocookie.com fonts.googleapis.com fonts.gstatic.com data: s.ytimg.com *.iubenda.com *.cloudfront.net; img-src * data:
```
Comment (aeneasr) [inline static/_headers:4]: Wouldn't we need to allow-list all analytics URLs etc here also?

edit:// yes, we get CSP violations

File: src/analytics.ts [hunk 2]
```diff
@@ -22,4 +32,106 @@ export const trackEvent = (event: IEvent) => {
...
+  var f = d.getElementsByTagName(s)[0],
+    j = d.createElement(s),
+    dl = l != 'dataLayer' ? '&l=' + l : ''
+  j.async = true
+  j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl
+  f.parentNode.insertBefore(j, f)
+}
+
+const loadGoogleAnalytics = () => {
+  initializeTagManger(window, document, 'script', 'dataLayer', 'GTM-5JC2SVK')
```
Comment (aeneasr) [inline src/analytics.ts:71]: @BlueAcidFrog can you please confirm that this ties into our existing GA account?

File: src/analytics.ts [hunk 3]
```diff
@@ -8,10 +8,20 @@ export interface IEvents {
   [key: string]: IEvent
 }
 
+enum ConsentCategory {
+  Necessary = '1',
+  Basic = '2',
+  Enhancement = '3',
+  Measurement = '4',
+  Advertising = '5'
```
Comment (aeneasr) [inline src/analytics.ts:16]: Does `Analytics`, `Enhancement`, `Basic` have an effect yet?

**Review by aeneasr — CHANGES_REQUESTED** [review]
A couple of to-dos:

1. [ ] Open incognito -> accept consent (reject all, accept all, ...) -> reload -> the banner shows up again
2. [ ] Missing CSP headers for gtag/ga/hubspot (see comment)
3. [ ] Ensure consent tracking works with ad blocker as well

Another question is that I believe we only load gtag and analytics but have all the categories enabled already. Do we need all of them?

File: src/analytics.ts (hunk 2 above)
Comment (BlueAcidFrog) [reply src/analytics.ts:71]: I can confirm the ID to be the correct one for Google Tag Manager (GTM-5JC2SVK). Ones we have set up GTM i can easily add GA Analytics to it.

File: src/analytics.ts (hunk 3 above)
Comment (MicLeey) [reply src/analytics.ts:16]: Not yet, they just correspond the granular options given by the consent manager

Comment (aeneasr) [reply src/analytics.ts:16]: I see, I assume those are then available in the tag manager for further processing?

File: src/analytics.ts (hunk 2 above)
Comment (aeneasr) [reply src/analytics.ts:71]: Nice, ty

**Comment by aeneasr** [comment]
@MicLeey i think we need to allow google analytics as well because gtag only adds the script which then loads google analytics

**Pull request merged**